app/feature/feature-terminate-insurance/src/main/kotlin/com/hedvig/android/feature/terminateinsurance/navigation/TerminateInsuranceGraph.kt

```diff
@@ -41,13 +39,23 @@ fun NavGraphBuilder.terminateInsuranceGraph(
   openChat: (NavBackStackEntry) -> Unit,
   openUrl: (String) -> Unit,
   openPlayStore: () -> Unit,
+  navigateToInsurances: (NavOptions) -> Unit,
   closeTerminationFlow: () -> Unit,
 ) {
   composable<TerminateInsuranceDestination.TerminationSuccess> { backStackEntry ->
     TerminationSuccessDestination(
       terminationDate = terminationDate,
       onSurveyClicked = { openUrl(surveyUrl) },
-      navigateBack = navigator::popBackStack,
+      onDone = {
+        if (!navController.popBackStack()) {
```

Comment:
That's a very good question. I think if it returns `false` it means that it failed to do anything at all, so it would fail to pop even this screen.
So we then instead do a navigate and we also make sure to pop `TerminateInsuranceDestination.TerminationSuccess`.
The good thing about this is that even if it works as you say, and it was popped anyway, then that popUpTo would be a no-op giving us the exact same end behavior. So all is good in either case